clustermesh-apiserver/etcdinit/root.go

```diff
@@ -154,14 +154,18 @@ func InitEtcdLocal() (returnErr error) {
 	defer func() {
 		log := log.WithField("etcdPID", etcdPid)
 		log.Debug("Cleaning up etcd process")
-		// Send the process a SIGTERM. SIGTERM is the "gentle" shutdown signal, and etcd should close down it's resources
+		// Send the process a SIGTERM. SIGTERM is the "gentle" shutdown signal, and etcd should close down its resources
 		// cleanly and then exit.
 		log.Info("Sending SIGTERM signal to etcd process")
 		err := etcdCmd.Process.Signal(syscall.SIGTERM)
 		if err != nil {
 			log.WithError(err).
 				Error("Failed to send SIGTERM signal to etcd process")
-			returnErr = err
+			// If we're not already returning an error, return an error here
+			if returnErr == nil {
+				returnErr = err
+			}
```

Comment:
Could we also use `errors.Join` here?

Reply:
I checked, and it does handle that case. I've updated all instances of it in that deferred function.